overlord/configstate/configcore/picfg.go

```diff
@@ -88,16 +89,35 @@ func updatePiConfig(path string, config map[string]string) error {
 	return nil
 }
 
-func piConfigFile(opts *fsOnlyContext) string {
+func piConfigFile(opts *fsOnlyContext) (string, error) {
 	rootDir := dirs.GlobalRootDir
+	subdir := "/boot/uboot"
 	if opts != nil {
 		rootDir = opts.RootDir
+	} else {
+		// not a filesystem only apply, so we may be operating on a run system
+		// on UC20, in which case we shouldn't use the /boot/uboot/ option and
+		// instead should use /run/mnt/ubuntu-seed/
+		kCmdlineVals, err := osutil.KernelCommandLineKeyValues("snapd_recovery_mode")
```

Comment:
why is this not using boot.ModeAndRecoverySystemFromKernelCommandLine ?

Reply:
sure I can refactor to use that instead